### PR TITLE
Update to use DependencyDescriptor

### PR DESCRIPTION
**Pull request by murphm8**
This is an update for an upcoming change to colcon-core: colcon/colcon-core#106

### CODE REVIEW

File: colcon_cargo/package_identification/cargo.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 
 from colcon_core.package_identification \
     import PackageIdentificationExtensionPoint
+from colcon_core.dependency_descriptor import DependencyDescriptor
```
Comment (dirk-thomas) [inline colcon_cargo/package_identification/cargo.py]: This line needs to be moved up by two to pass the `flake8-import-order` check.

File: colcon_cargo/package_identification/cargo.py [hunk 2]
```diff
@@ -33,8 +34,11 @@ def identify(self, metadata):  # noqa: D102
         metadata.type = 'cargo'
         if metadata.name is None:
             metadata.name = data['name']
-        metadata.dependencies['build'] |= data['depends']
-        metadata.dependencies['run'] |= data['depends']
+
+        dependencies = set(DependencyDescriptor(name)
+                           for name in data['depends'])
+        metadata.dependencies['build'] |= dependencies
+        metadata.dependencies['run'] |= dependencies
```
Comment (dirk-thomas) [inline colcon_cargo/package_identification/cargo.py]: With this patch both dependencies are sharing the same `DependencyDescriptor` instance. I would suggest to use separate ones instead:

```
metadata.dependencies['build'] |= {
    DependencyDescriptor(name) for name in data['depends']}
metadata.dependencies['run'] |= {
    DependencyDescriptor(name) for name in data['depends']}
```

**Comment by dirk-thomas** [comment]
The patch looks good to me. We should hold it (see https://github.com/colcon/colcon-core/pull/107#pullrequestreview-154454998).

At the end I will leave it up the maintainer of this package to merge this when ready.

**Comment by murphm8** [comment]
Since this doesn't add any useful metadata this can be closed for now. If there is metadata to add in the future this package can be updated to use DependencyDescriptor and add the metadata at that time. Closing.

**Pull request closed without merging**